package.json

```diff
@@ -43,9 +44,15 @@
     "lint": "esw --color src --ext .ts",
     "fix": "yarn lint --fix",
     "eslint-check": "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
-    "build": "tsc",
+    "build": "rm -rf lib/* && tsc && tsc -p tsconfig.esm.json",
     "watch": "tsc --watch"
   },
+  "sideEffects": false,
+  "exports": {
+    "import": "./lib/esm/index.js",
+    "require": "./lib/cjs/index.js"
+  },
+  "type": "module",
```

Comment:
Be careful with this. AFAIK this is wrong. This library exports both ESM and CJS builds. With `type: module`, you are informing Node that this library exports an ESM build which is not the case.

I would simply remove this line here. Let the client decide whether he wants to choose the ESM or CJS build.

For reference: https://nodejs.org/docs/latest-v13.x/api/esm.html#esm_enabling